instrumentation/opentelemetry-instrumentation-openai/src/opentelemetry/instrumentation/openai/utils.py

```diff
@@ -0,0 +1,184 @@
+
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except Exception as exception:
+            logger.warning(
+                "Failed to execute %s, error: %s",
+                func.__name__,
+                str(exception),
+            )
+
+    return wrapper
+
+
+def extract_content(choice):
+    if getattr(choice, "message", None) is None:
+        return ""
+
+    # Check if choice.message exists and has a content attribute
+    message = choice.message
+    if getattr(message, "content", None):
+        return choice.message.content
+
+    # Check if choice.message has tool_calls and extract information accordingly
+    elif getattr(message, "tool_calls", None):
+        result = [
+            {
+                "id": tool_call.id,
+                "type": tool_call.type,
+                "function": {
+                    "name": tool_call.function.name,
+                    "arguments": tool_call.function.arguments,
+                },
+            }
+            for tool_call in choice.message.tool_calls
+        ]
+        return result
+
+    # Check if choice.message has a function_call and extract information accordingly
+    elif getattr(message, "function_call", None):
+        return {
+            "name": choice.message.function_call.name,
+            "arguments": choice.message.function_call.arguments,
+        }
+
+    # Return an empty string if none of the above conditions are met
+    else:
+        return ""
+
+
+def extract_tools_prompt(item):
+    tool_calls = getattr(item, "tool_calls", None)
+    if tool_calls is None:
+        return
+
+    calls = []
+    for tool_call in tool_calls:
+        tool_call_dict = {
+            "id": getattr(tool_call, "id", ""),
+            "type": getattr(tool_call, "type", ""),
+        }
+
+        if hasattr(tool_call, "function"):
+            tool_call_dict["function"] = {
+                "name": getattr(tool_call.function, "name", ""),
+                "arguments": getattr(tool_call.function, "arguments", ""),
+            }
+        calls.append(tool_call_dict)
+    return calls
+
+
+def set_event_prompt(span: Span, prompt):
+    span.add_event(
+        name="gen_ai.content.prompt",
+        attributes={
+            GenAIAttributes.GEN_AI_PROMPT: prompt,
+        },
+    )
+
+
+def set_span_attributes(span: Span, attributes: dict):
+    for field, value in attributes.model_dump(by_alias=True).items():
+        set_span_attribute(span, field, value)
+
+
+def set_event_completion(span: Span, result_content):
+    span.add_event(
+        name="gen_ai.content.completion",
+        attributes={
+            GenAIAttributes.GEN_AI_COMPLETION: json.dumps(result_content),
+        },
+    )
+
+
+def set_span_attribute(span: Span, name, value):
+    if non_numerical_value_is_set(value) is False:
+        return
+
+    if name == GenAIAttributes.GEN_AI_PROMPT:
+        set_event_prompt(span, value)
```

Comment:
it's not an attribute, so could you please handle it separately from this method?